corehq/apps/locations/tests/test_permissions.py

```diff
@@ -243,3 +241,57 @@ def test_user_can_acces_case(self):
                                            owner_id=self.locations['Cambridge'].location_id)
         self.assertTrue(user_can_access_case(self.domain, self.cambridge_worker, self.case))
         self.assertFalse(user_can_access_case(self.domain, self.suffolk_user, self.case))
+
+
+class TestLocationExport(LocationHierarchyTestCase):
+    domain = 'test-location-export'
+    location_type_names = ['state', 'county', 'city']
+    stock_tracking_types = ['state', 'county', 'city']
+    location_structure = [
+        ('England', [
+            ('Cambridgeshire', [
+                ('Cambridge', []),  # The other Cambridge ;)
+                ('Peterborough', []),
+            ]),
+            ('Lincolnshire', [
+                ('Boston', []),  # The other Boston ;)
+            ])
+        ]),
+        ('California', [
+            ('Los Angeles', []),
+        ])
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.user = WebUser.create(cls.domain, 'jbloggs', 'Passw0rd!', None, None)
+        cls.user.set_location(cls.domain, cls.locations['Cambridgeshire'])
+        cls.restrict_user_to_assigned_locations(cls.user)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete(cls.domain, deleted_by=None)
+        super().tearDownClass()
+
+    def test_location_filter_form(self):
+        """
+        ``LocationFilterForm.is_valid()`` returns True although ``user``
+        is in a separate location hierarchy from ``location_id``.
+        """
+        request_params = {
+            'location_id': self.locations['California'].location_id,
+            'selected_location_only': False,
+            'location_status_active': LocationFilterForm.SHOW_ALL,
+        }
+        form = LocationFilterForm(
+            request_params,
+            domain=self.domain,
+            user=self.user,
+        )
+        self.assertFalse(form.is_valid())
```

Comment:
It would be great to add a test for the inverse, where a non-location restricted user can still access this.

Reply:
59b1c014a7fc9b5928fe7f206703cb52fdd49573